File: scala/src/main/scala/com/laamella/code_state_machine/condition/AlwaysCondition.scala
package com.laamella.code_state_machine.condition

/**
 * This condition is always met.
 */
final class AlwaysCondition[E] extends NonEventBasedCondition[E] {
	/** @return whether the condition is met. */
	override def isMet: Boolean = true

	override def toString = "always"
}

File: java/src/main/java/com/laamella/code_state_machine/condition/AlwaysCondition.java
package com.laamella.code_state_machine.condition;

/**
 * This condition is always met.
 */
public final class AlwaysCondition<E> extends NonEventBasedCondition<E> {
	@Override
	public boolean isMet() {
		return true;
	}

	@Override
	public String toString() {
		return "always";
	}
}